File: src/main/java/com/raychenon/hackerrank/AVeryBigSum.java
package com.raychenon.hackerrank;


/**
 * User: raychenon
 * Date: 13/1/19
 * https://www.hackerrank.com/challenges/a-very-big-sum/problem
 */
public class AVeryBigSum {
    
    /**
     * time complexity : O(n)
     * space complexity :  O(1)
     * @param ar
     * @return sum
     */
    static long aVeryBigSum(long[] ar) {
        long sum = 0;
        for (int i = 0; i < ar.length; i++) {
            sum += ar[i];
        }
        return sum;
    }

}

File: python3/hackerrank/a_very_big_sum.py
#!/bin/python3

"""
https://www.hackerrank.com/challenges/a-very-big-sum/problem
"""


def aVeryBigSum(ar):
    sum = 0
    for e in ar:
        sum = sum + e
    return sum


def test_sum_single_element():
    assert aVeryBigSum([1]) == 1


def test_sum():
    assert aVeryBigSum([4, 5]) == 9


def test_larger_sum():
    assert aVeryBigSum([1000000001, 1000000002, 1000000003, 1000000004, 1000000005]) == 5000000015
